prefect_census/runs.py

```diff
@@ -0,0 +1,198 @@
+                    - connection_id
+                    - object
+                - mappings
+                    - from
+                    - type
+                    - data
+                    - to
+                    - is_primary_identifier
+                    - generate_field
+                    - preserve_values
+                    - operation
+
+
+    Example:
+        Get Census sync run info:
+        ```python
+        from prefect import flow
+
+        from prefect_census import CensusCredentials
+        from prefect_census.runs import get_census_sync_run_info
+
+        @flow
+        def get_sync_run_info_flow():
+            credentials = CensusCredentials(api_key="my_api_key")
+
+            return get_census_sync_run_info(
+                credentials=credentials,
+                run_id=42
+            )
+
+        get_sync_run_info_flow()
+        ```
+    """  # noqa
+    try:
+        async with credentials.get_client() as client:
+            response = await client.get_run_info(run_id)
+    except HTTPStatusError as e:
+        raise CensusGetSyncRunInfoFailed(extract_user_message(e)) from e
+
+    return response.json()["data"]
+
+
+@flow(
+    name="Wait for Census sync run",
+    description="Waits for the Census sync run to finish running.",
+)
+async def wait_census_sync_completion(
+    run_id: int,
+    credentials: CensusCredentials,
+    max_wait_seconds: int = 60,
+    poll_frequency_seconds: int = 5,
+) -> Tuple[CensusSyncRunStatus, Dict[str, Any]]:
+    """
+    Wait for the given Census sync run to finish running.
+
+    Args:
+        run_id: The ID of the sync run to wait for.
+        credentials: Credentials for authenticating with Census.
+        max_wait_seconds: Maximum number of seconds to wait for sync to complete.
+        poll_frequency_seconds: Number of seconds to wait in between checks for
+            run completion.
+
+    Raises:
+        CensusSyncRunTimeout: When the elapsed wait time exceeds `max_wait_seconds`.
+
+    Returns:
+        run_status: An enum representing the final Census sync run status.
+        run_data: A dictionary containing information about the run after completion.
+    """
+    logger = get_run_logger()
+    seconds_waited_for_run_completion = 0
+    wait_for = []
+    while seconds_waited_for_run_completion <= max_wait_seconds:
+        run_data_future = await get_census_sync_run_info.submit(
+            credentials=credentials,
+            run_id=run_id,
+            wait_for=wait_for,
+        )
+        run_data = await run_data_future.result()
+        run_status = run_data.get("status")
+
+        if CensusSyncRunStatus.is_terminal_status_code(run_status):
+            return CensusSyncRunStatus(run_status), run_data
+
+        wait_for = [run_data_future]
+        logger.info(
+            "Census sync run with ID %i has status %s. Waiting for %i seconds.",
+            run_id,
+            CensusSyncRunStatus(run_status),
```

Comment:
This makes the log status a bit easier to read.
```suggestion
            CensusSyncRunStatus(run_status).name,
```